sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ClientSideRequestStatistics.java

```diff
@@ -127,15 +128,17 @@ void recordGatewayResponse(
                 this.requestEndTime = responseTime;
             }
 
-            if (rxDocumentServiceRequest != null 
-                    && rxDocumentServiceRequest.requestContext != null 
+            if (rxDocumentServiceRequest != null
+                    && rxDocumentServiceRequest.requestContext != null
                     && rxDocumentServiceRequest.requestContext.retryContext != null) {
                 rxDocumentServiceRequest.requestContext.retryContext.retryEndTime = ZonedDateTime.now(ZoneOffset.UTC);
                 this.retryContext = new RetryContext(rxDocumentServiceRequest.requestContext.retryContext);
             }
 
             this.gatewayStatistics = new GatewayStatistics();
-            this.gatewayStatistics.operationType = rxDocumentServiceRequest.getOperationType();
+            if (rxDocumentServiceRequest != null) {
```

Comment:
@simplynaveen20 to review request diagnostics changes.

Reply:
done